File: cpp/data_structures/four_chapter/decimalConvertother.c
#include <iostream>
#include <stack>

using namespace std;


void convert(stack<char>& s, int n, int base) {
	static char digit[] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F'};
	while (n > 0) {
		s.push(digit[n % base]);
		n /= base;
	}
}

int main(int argc, char** argv)
{
	int n, base;
	static stack<char> s;
	while (cout << "please input number and base:", cin >> n >> base) {
		cout << "the number of " << base << " is: ";
		convert(s, n, base);
		while (!s.empty()) {
			cout << s.top();
			s.pop();
		}
		cout << endl;
	}
				 
	

	return 0;
}

File: go/data_structure/4_chapter/decimalConvertOther.go
package main

import (
	"container/list"
	"fmt"
	"log"
)

func main() {
	var n, b int
	var s list.List
	for true {
		fmt.Println("please input number and base:")
		_, err := fmt.Scan(&n, &b)
		if err != nil {
			log.Fatal(err)

		}
		fmt.Printf("the number of %d base is: 0x", b)
		convert(&s, n, b)
		for s.Len() != 0 {
			fmt.Printf("%c", s.Back().Value)
			s.Remove(s.Back())
		}
		fmt.Println()
	}
}

var digit = []byte{'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'}

func convert(s *list.List, n, b int) {
	for n > 0 {
		s.PushBack(digit[n%b])
		n /= b
	}
}
